R/tbl_split.R

```diff
@@ -2,22 +2,47 @@
 #'
 #' `r lifecycle::badge('experimental')`\cr
 #' The `tbl_split` function splits a single gtsummary table into multiple tables.
+#' Both horizontal (column-wise, (that is, splits by columns in
+#' `x$table_body`) and vertical (row-wise) splits are possible.
 #' Updates to the print method are expected.
 #'
-#' @param x (`gtsummary`)\cr
-#'   gtsummary table
+#' @param x (`gtsummary` or `list`)\cr
+#'   gtsummary table.
 #' @param variables ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
 #'   variables at which to split the gtsummary table rows (tables
-#'   will be separated after each of these variables)
+#'   will be separated after each of these variables).
+#' @param keys ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns to be repeated in each table split.
+#' @param groups (list of `character` vectors)\cr
+#'   list of column names that appear in `x$table_body`.
+#'   Each group of column names represent.
 #' @inheritParams rlang::args_dots_empty
 #'
 #' @return `tbl_split` object
-#' @name tbl_split
+#'
+#' @details
+#' Run `gtsummary::show_header_names()` to print all column names to split by.
 #'
 #' @examples
-#' tbl <-
-#'   tbl_summary(trial) |>
+#' # create standard table
+#' tbl <- tbl_summary(trial, by = trt)
+#'
+#' # split by rows
+#' tbl_list_rows <- tbl |>
 #'   tbl_split(variables = c(marker, grade))
+#'
+#' # print column names
+#' gtsummary::show_header_names(tbl)
+#'
+#' # split by columns
+#' tbl_list_cols <- tbl |>
+#'   tbl_split(key = c(stat_1, stat_2))
+#'
+#' # take the list of splitted rows and split again by columns
+#' tbl_list_rows_cols <- tbl_list_rows |>
+#'   tbl_split(key = c(stat_1, stat_2))
+#'
+#' @name tbl_split
```

Comment:
I moved it here bc I always think it is a bit clearer than having to search for it in the doc above. Feel free to move it back there